### PR TITLE
new style `batch_l2_norm_squared` function

### PR DESCRIPTION
**Pull request by kmaehashi**
This PR implements new style `batch_l2_norm_squared` function.

### CODE REVIEW

**Comment by kmaehashi** [comment]
Confirmed to pass 10,000 repeats.

File: chainer/functions/math/batch_l2_norm_squared.py [hunk 1]
```diff
@@ -18,29 +21,51 @@ def check_type_forward(self, in_types):
...
+            'l2normsquared_bwd')
+        gx = kernel(x, gy0)
+        return gx,
+
+    def backward(self, indexes, grad_outputs):
+        x, gy0 = self.get_retained_inputs()
+        gy0 = gy0.reshape(-1, *((1,) * (x.ndim - 1)))
+        gy0 = chainer.functions.broadcast_to(gy0, x.shape)
+        gg2 = 2 * grad_outputs[0]
+        ggx = gg2 * gy0
```
Comment (takagi) [inline chainer/functions/math/batch_l2_norm_squared.py]: The name `gx` would better than `ggx` here because we want the gradient of `x`.

Comment (kmaehashi) [reply chainer/functions/math/batch_l2_norm_squared.py]: Absolutely. Fixed.

**Comment by takagi** [comment]
jenkins, test this please.

**Pull request merged**